### PR TITLE
[rocm-opencl-runtime] Enable support for gfx800

### PR DESCRIPTION
**Pull request by John-Gee**
Hello,

As discussed in #741, this should have minimum impact outside of enabling gfx800 cards again.

I'm open to any change done by me or you of course.

Thank you!

### CODE REVIEW

**Comment by John-Gee** [comment]
Should this somehow be marked as unsupported by AMD though?

**Comment by JP-Ellis** [comment]
While you included the name and hash of the patch, you didn't commit the file nor did you include the URL where it came from.  I'm guessing you used [this patch file](https://github.com/xuhuisheng/rocm-build/blob/feature/build/patch/31.rocm-opencl-runtime-rocclr-gfx803-1.patch)?

Probably best to include the file directly in the commit along with a comment describing where it came from.  This will prevent upstream changes in the patch file from breaking the build, but still makes it easy to find the new relevant patch.

**Comment by John-Gee** [comment]
Woops, my apologies, and I actually checked that the PR looked fine but didn't realize the file was missing... Sorry!

I actually had already done the other part, I hope that's the way you were thinking.
Thanks for this!

**Comment by acxz** [comment]
> Should this somehow be marked as unsupported by AMD though?

No need, even though many of our users try to use unsupported ROCm cards, we still try to help & debug & contribute based on them.

We'll leave the issue open with the patched label to denote we are doing something different from upstream AMD, however.

**Review by acxz — CHANGES_REQUESTED** [review]
Can you update the [pkgrel](https://wiki.archlinux.org/title/PKGBUILD#pkgrel) and the [.SRCINFO](https://wiki.archlinux.org/title/.SRCINFO) file?
See: https://github.com/rocm-arch/rocm-arch/#contributing

**Comment by John-Gee** [comment]
Done!

Thank you!

File: rocm-opencl-runtime/PKGBUILD [hunk 1]
```diff
@@ -15,12 +15,19 @@ provides=('opencl-driver')
...
 sha256sums=('362d81303048cf7ed5d2f69fb65ed65425bc3da4734fff83e3b8fbdda51b0927'
-            'f4f265604b534795a275af902b2c814f416434d9c9e16db81b3ed5d062187dfa')
+            'f4f265604b534795a275af902b2c814f416434d9c9e16db81b3ed5d062187dfa'
+            'b186dd4a604d6e8a2c94ba6569638aaa8066558d764aa0d9cf76f998724ed90a')
 _dirname="$(basename "$url")-$(basename "${source[0]}" .tar.gz)"
 _rocclr_dir="$(basename "$_rocclr")-$(basename "${source[1]}" .tar.gz)"
 
+prepare() {
+	cd "ROCclr-rocm-$pkgver"
+	patch -p1 -i "$srcdir/enable-gfx800.patch"
```
Comment (tpkessler) [inline rocm-opencl-runtime/PKGBUILD:28]: Could you please add a link to the relevant discussion upstream @JP-Ellis suggested?

**Comment by tpkessler** [comment]
Thanks for working on this @John-Gee! I will merge this after you will have taken care of the comments above.

**Comment by John-Gee** [comment]
Hey,

I am not on my computer so I'll do the changes later, but isn't the link in
the patch enough or do you want it in both the patch and PKGBUILD or just
the latter?

Thanks for your help!

**Comment by tpkessler** [comment]
LGTM! Thank you very much @John-Gee.

**Comment by John-Gee** [comment]
Thank you all!

**Pull request merged**